tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -1103,7 +1125,52 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
 
     std::vector<HloInstruction *> operands(gemm->operands().begin(),
                                            gemm->operands().end());
-    operands.insert(operands.begin() + 2, MaybeConstantFoldBias(bias));
+    HloInstruction *broadcast_bias = MaybeConstantFoldBias(bias);
+    if (slice) {
+      broadcast_bias = instr->AddInstruction(
+          HloInstruction::CreateBitcast(slice->shape(), broadcast_bias));
+    }
+    absl::Span<const int64_t> batch_dims =
+        config.dot_dimension_numbers().rhs_batch_dimensions();
+    // Get the padded shape.
+    auto pad_shape = [&batch_dims](const Shape old_shape) {
```

Comment:
Does this duplicate functionality from CreateF8CustomCall? Can the lambda be replaced by a function?

Reply:
Placed by functions in newer commit.